### PR TITLE
Removal of in-repo documentation

### PR DESCRIPTION
**Pull request by theletterf**
- Removes most in-repo documentation
- Links to official documentation using QuickDraw links, to facilitate link updates later on
- Edits README file
- Preserves metrics and settings docs, pending automation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,80 +81,29 @@ The following documentation refers to the in-development version of `splunk-otel
...
-
-> **Note**: Supported java servers may require different steps to add the path to the JVM agent to their configuration. See [Add the JVM agent to Java servers](./docs/server-instructions.md).
-
-> **Tip**: To generate a snippet that includes all the basic install commands for your environment and service, open the Splunk Observability Cloud wizard in **Data Setup > APM Instrumentation > Java > Add Connection**.
-
-When you run your application with the Java agent, trace data goes to Observability Cloud through the Splunk OTel collector. If no data
-appears in **Observability > APM**, see [Troubleshooting](#troubleshooting).
-
-For full install instructions, including K8s and Lambda snippets, see [Instrument Java services for Observability Cloud](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/get-started.html) in the official Splunk documentation.
+For complete instructions on how to get started with the Splunk Distribution of OpenTelemetry JS, see [Instrument Java services for Observability Cloud](https://quickdraw.splunk.com/redirect/?product=Observability&version=current&location=learnmore.java.gdi) in the official Splunk documentation.
```
Comment (mateuszrzeszutek) [inline README.md:88]:
```suggestion
For complete instructions on how to get started with the Splunk Distribution of OpenTelemetry Java, see [Instrument Java services for Observability Cloud](https://quickdraw.splunk.com/redirect/?product=Observability&version=current&location=learnmore.java.gdi) in the official Splunk documentation.
```

File: docs/contributing/metrics.md
Comment (mateuszrzeszutek) [inline docs/contributing/metrics.md:1]: Please add this file back -- it actually contains technical documentation about the metrics feature, targeted to the contributors (us).

File: docs/advanced-config.md
Comment (mateuszrzeszutek) [inline docs/advanced-config.md:1]: This file seems to duplicate the main docs; why don't we remove it too?

File: docs/README.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
-The official documentation for this distribution can be found here:
+# The official documentation for this distribution can be found here:
```
Comment (mateuszrzeszutek) [inline docs/README.md:1]: I'm not sure whether this actually should be a heading

File: README.md [hunk 3]
```diff
@@ -175,13 +122,41 @@ We publish [snapshot builds](https://oss.sonatype.org/content/repositories/snaps
...
+## Understanding version numbers
+
+Refer to
+the [versioning document](https://github.com/signalfx/splunk-otel-java/blob/main/VERSIONING.md) to
+learn more about version numbers. Major versions contain a large number of changes, which might
+result in increased risk to your production environment. The most common releases are marked with a
+minor version, and they contain a modest number of changes. Patch releases are infrequent, and they
+pinpoint specific fixes or enhancements.
 
-For information and best practices around upgrades, see the [Upgrading documentation](docs/upgrading.md).
```
Comment (mateuszrzeszutek) [inline README.md:180]: I think we've lost some of the useful info from the upgrading doc: watching for new releases, how ofter we release, and that you should verify the agent just as any other software component.

File: docs/README.md (hunk 2 above)
Comment (theletterf) [reply docs/README.md:1]: It's for formatting / visibility, mainly.

File: README.md (hunk 3 above)
Comment (theletterf) [reply README.md:180]: Let's discuss this one.

File: docs/contributing/metrics.md
Comment (theletterf) [reply docs/contributing/metrics.md:1]: Done!

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**